apps/research-service/src/validations/validation.service.ts

```diff
@@ -1,13 +1,25 @@
 import { BadRequestException, Injectable, NotFoundException } from "@nestjs/common";
-import { CriteriaSite, PolygonGeometry, SitePolygon } from "@terramatch-microservices/database/entities";
+import {
+  CriteriaSite,
+  CriteriaSiteHistoric,
+  PolygonGeometry,
+  SitePolygon
+} from "@terramatch-microservices/database/entities";
 import { ValidationDto } from "./dto/validation.dto";
-import { ValidationCriteriaDto } from "./dto/validation-criteria.dto";
+import { ValidationRequestDto } from "./dto/validation-request.dto";
+import { ValidationResponseDto, ValidationCriteriaDto } from "./dto/validation-criteria.dto";
 import { populateDto } from "@terramatch-microservices/common/dto/json-api-attributes";
 import { MAX_PAGE_SIZE } from "@terramatch-microservices/common/util/paginated-query.builder";
 import { groupBy } from "lodash";
+import { SelfIntersectionValidator } from "./validators/self-intersection.validator";
+import { SpikesValidator } from "./validators/spikes.validator";
 
 @Injectable()
 export class ValidationService {
+  constructor(
+    private readonly selfIntersectionValidator: SelfIntersectionValidator,
+    private readonly spikesValidator: SpikesValidator
+  ) {}
```

Comment:
I don't see any reason for these to be injectable - they don't have any internal dependencies on other services. I think a better pattern would be to just make a mapping of available validators a the top like we do in `airtable.processor.ts` or `entities.service.ts`